src/components/composite/common/CategorySelector/CategorySelector.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  alpha,
+  Box,
+  List,
+  ListItemButton,
+  ListItemIcon,
+  ListItemProps,
+  ListItemText,
+  styled,
+  Typography,
+} from '@mui/material';
+import React, { ComponentType, FC, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export interface CategoryConfig {
+  title: string;
+  icon?: ComponentType;
+}
+
+interface CategorySelectorProps {
+  categories: CategoryConfig[];
+  value?: string;
+  showLabels?: boolean;
+  onSelect?: (category: string) => void;
+}
+
+export const CATEGORY_ALL = 'Show all';
```

Comment:
Is it required that "show all" option is handled from within the selector component?
It looks like a normal option to me, so if it's special in the context of the parent component, that should be addressed there. One benefit of handling "show all" in a parent would be possibility to create category selectors without "show all" with no extra work.
(Another one would be less coupling between components - now you have to export "show all" and import it in the parent).